### PR TITLE
test(Settings): adds a test that sends a contact request with chat key

### PR DESCRIPTION
**Pull request by jrainville**
Partial fix of #6883 

This sends a contact request using the chat key

I'll create a new PR that uses a chat message to send a request

### CODE REVIEW

**Comment by jrainville** [comment]
Please review when you can :pray:

File: test/ui-test/src/screens/SettingsScreen.py [hunk 1]
```diff
@@ -343,5 +351,25 @@ def change_user_password(self, oldPassword: str, newPassword: str):
...
+        
+        click_obj_by_name(ContactsViewScreen.CONTACT_REQUEST_SEND_BUTTON.value)
+
+    def verify_contact_request(self, chat_key: str):
+        time.sleep(0.5)
+        click_obj_by_name(ContactsViewScreen.CONTACT_REQUEST_PENDING_REQUEST_TAB_BUTTON.value)
+        contact_list = get_obj(ContactsViewScreen.SENT_REQUESTS_CONTACT_PANEL_LIST_VIEW.value)
+        for index in range(contact_list.count):
+            contact = contact_list.itemAtIndex(index)
+            print("Found: ", contact.compressedPk, ", Wanted:  ", chat_key)
```
Comment (stefandunca) [inline test/ui-test/src/screens/SettingsScreen.py:370]: Is this a leftover or do we want to have "logging" into the std output?

**Review by stefandunca — APPROVED** [review]
Looks and works great!

image

File: test/ui-test/src/screens/SettingsScreen.py (hunk 1 above)
Comment (jrainville) [reply test/ui-test/src/screens/SettingsScreen.py:370]: I wanted to leave it, since the verify failure at the bottom doesn't tell us much info if it fails

Comment (stefandunca) [reply test/ui-test/src/screens/SettingsScreen.py:370]: Relying on std out for general logging generates noise after more and more output is added to it. I'm thinking that Maybe a more verbose verify will help improve the fail message and don't add up to become noisy output; e.g. `verify_failure(f'The found contact chat key "{contact.compressedPk}" doesn't match wanted chat key "{chat_key}" in the list of pending contacts')`

File: test/ui-test/src/screens/SettingsScreen.py [hunk 2]
```diff
@@ -51,7 +51,15 @@ class MessagingOptionScreen(Enum):
...
-
+    CONTACTS_BTN: str = "contacts_listItem_btn"
+   
+class ContactsViewScreen(Enum):
+    CONTACT_REQUEST_CHAT_KEY_BTN: str = "contact_request_to_chat_key_btn"
+    CONTACT_REQUEST_CHAT_KEY_INPUT: str = "contactRequest_ChatKey_Input"
+    CONTACT_REQUEST_SAY_WHO_YOU_ARE_INPUT: str = "contactRequest_SayWhoYouAre_Input"
+    CONTACT_REQUEST_SEND_BUTTON: str = "contactRequest_Send_Button"
+    CONTACT_REQUEST_PENDING_REQUEST_TAB_BUTTON: str = "contactRequest_PendingRequests_Button"
+    SENT_REQUESTS_CONTACT_PANEL_LIST_VIEW: str = "sentRquests_contactListPanel_ListView"
```
Comment (noeliaSD) [inline test/ui-test/src/screens/SettingsScreen.py:62]:
```suggestion
    SENT_REQUESTS_CONTACT_PANEL_LIST_VIEW: str = "sentRequests_contactListPanel_ListView"
```

File: test/ui-test/src/screens/SettingsScreen.py [hunk 3]
```diff
@@ -343,5 +351,25 @@ def change_user_password(self, oldPassword: str, newPassword: str):
...
+        click_obj_by_name(MessagingOptionScreen.CONTACTS_BTN.value)
+        click_obj_by_name(ContactsViewScreen.CONTACT_REQUEST_CHAT_KEY_BTN.value)
+        
+        type(ContactsViewScreen.CONTACT_REQUEST_CHAT_KEY_INPUT.value, chat_key)
+        type(ContactsViewScreen.CONTACT_REQUEST_SAY_WHO_YOU_ARE_INPUT.value, who_you_are)
+        
+        click_obj_by_name(ContactsViewScreen.CONTACT_REQUEST_SEND_BUTTON.value)
+
+    def verify_contact_request(self, chat_key: str):
+        time.sleep(0.5)
```
Comment (noeliaSD) [inline test/ui-test/src/screens/SettingsScreen.py:365]: Could we avoid this???

File: test/ui-test/testSuites/suite_settings/tst_contactsFlow/test.feature [hunk 4]
```diff
@@ -0,0 +1,16 @@
+Feature: Status Desktop Contacts Flows
+
+    As a user I want to login the app and interact with contacts (add, remove, etc)
+
+    Background: Sign up and open settings section
+        Given A first time user lands on the status desktop and generates new key
+        When user signs up with username "tester123" and password "TesTEr16843/!@00"
```
Comment (noeliaSD) [inline test/ui-test/testSuites/suite_settings/tst_contactsFlow/test.feature:7]:
```suggestion
        And user signs up with username "tester123" and password "TesTEr16843/!@00"
```

File: test/ui-test/testSuites/suite_settings/tst_contactsFlow/test.feature [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+        Then the user lands on the signed in app
```
Comment (noeliaSD) [inline test/ui-test/testSuites/suite_settings/tst_contactsFlow/test.feature:8]:
```suggestion
        And the user lands on the signed in app
```

File: test/ui-test/testSuites/suite_settings/tst_contactsFlow/test.feature [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+        When the user opens app settings screen
```
Comment (noeliaSD) [inline test/ui-test/testSuites/suite_settings/tst_contactsFlow/test.feature:9]:
```suggestion
        And the user opens app settings screen
```

File: test/ui-test/testSuites/suite_settings/tst_contactsFlow/test.feature [hunk 7]
```diff
@@ -0,0 +1,16 @@
+Feature: Status Desktop Contacts Flows
```
Comment (noeliaSD) [inline test/ui-test/testSuites/suite_settings/tst_contactsFlow/test.feature:1]: @glitchminer could you take a look at this new feature and scenario? Thanks!

**Review by noeliaSD — APPROVED** [review]
LGTM!!

Just added minor comments and requested @glitchminer review for the new BDD statements!

**Comment by jrainville** [comment]
@stefandunca @noeliaSD I updated according to your comments. Thanks.

@glitchminer can you review please :)

**Review by noeliaSD — APPROVED** [review]

**Pull request merged**